cloudflare/resource_cloudflare_zone_dnssec.go

```diff
@@ -0,0 +1,155 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// The supported status for a Zone DNSSEC setting
+const (
+	DNSSECStatusActive   = "active"
+	DNSSECStatusDisabled = "disabled"
+)
+
+func resourceCloudflareZoneDNSSEC() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareZoneDNSSECCreate,
+		Read:   resourceCloudflareZoneDNSSECRead,
+		Update: resourceCloudflareZoneDNSSECUpdate,
+		Delete: resourceCloudflareZoneDNSSECDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"flags": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"algorithm": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"key_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"digest_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"digest_algorithm": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"digest": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ds": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"key_tag": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"public_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified_on": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareZoneDNSSECCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	zoneID := d.Get("zone_id").(string)
+
+	log.Printf("[INFO] Creating Cloudflare Zone DNSSEC: name %s", zoneID)
+
+	_, err := client.UpdateZoneDNSSEC(zoneID, cloudflare.ZoneDNSSECUpdateOptions{Status: DNSSECStatusActive})
+
+	if err != nil {
+		return fmt.Errorf("Error creating zone DNSSEC %q: %s", zoneID, err)
+	}
+
+	d.SetId(zoneID)
+
+	return resourceCloudflareZoneDNSSECRead(d, meta)
+}
+
+func resourceCloudflareZoneDNSSECRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	zoneID := d.Id()
```

Comment:
As we are in the `Read` method, this should come from the schema as the ID may not always be set at this point.

```suggestion
	zoneID := d.Get("zone_id").(string)
```